### ISSUE TITLE
VS code outputs low-level error when rejecting commands

### ISSUE THREAD

**Issue by matthiasS-da** [digital-asset/daml#5903]
Examples error messages:
`Scenario service backend error: BErrorClient (ClientIOError (GRPCIOBadStatusCode StatusUnknown (StatusDetails {unStatusDetails = \"\"})))`
`CRASH: Key com.daml.lf.transaction.Node$GlobalKey@6a396b63 not found`

Expected behavior: The error message says why the command has failed. (E.g., contract not active, contract not visible.)

Version: 1.1.0-snapshot.20200430.4057.0.681c862d

Detailed example:
```
template Foo
  with
    m : Party
    o : Party
  where
    signatory m
    observer o
    key m : Party
    maintainer key

-- Use this to dislose a Foo with signatory s to a party r
template DiscloseFoo 
  with
    s : Party
    r : Party
  where
    signatory s
    observer r

    controller s can
      Disclose : ContractId Foo
        do
          (fooId, _) <- fetchByKey @Foo s
          return fooId

-- Contract where s authorizes arbitrary parties to fetch Foo contracts
template Baz
  with 
    s : Party
    o : Party
  where
    signatory s
    observer o

    nonconsuming choice BazFoo : ()
      with
        c : Party
        m : Party
      controller c
        do
          (fooId, _) <- fetchByKey @Foo m
          return ()

technical = scenario do
  alice <- getParty "Alice"
  bob <- getParty "Bob"
  
  aliceFooId <- submit alice do create Foo with m = alice, o = bob
  bobFooId <- submit bob do create Foo with m = bob, o = bob

  -- Now let's do some delegation
  aliceBazId <- submit alice do create Baz with s = alice, o = bob
  bobBazId <- submit bob do create Baz with s = bob, o = alice

  -- Succeeds as expected because all involved contracts belong to alice
  _ <- submit alice do exercise aliceBazId BazFoo with c = alice, m = alice

  -- Should fail because alice cannot see bobFooId.
  -- FIXME: Fails with a low level error:
  -- Scenario service backend error: BErrorClient (ClientIOError (GRPCIOBadStatusCode StatusUnknown (StatusDetails {unStatusDetails = \"\"})))
  _ <- submitMustFail alice do exercise aliceBazId BazFoo with c = alice, m = bob

  -- Fails because alice cannot see bobFooId.
  -- FIXME: Again a low-level error
  _ <- submitMustFail alice do exercise bobBazId BazFoo with c = alice, m = bob

  -- Disclose bobFooId to alice
  discloseId <- submit bob do create DiscloseFoo with s = bob, r = alice
  _ <- submit bob do exercise discloseId Disclose

  -- Fails because alice is not a maintainer of bobFooId
  _ <- submitMustFail alice do exercise aliceBazId BazFoo with c = alice, m = bob

  -- Now succeeds
  _ <- submit alice do exercise bobBazId BazFoo with c = alice, m = bob

  -- Now archive the contract.
  _ <- submit bob do archive bobFooId

  -- Should fail because the fetched contract is inactive.
  -- FIXME: CRASH: Key com.daml.lf.transaction.Node$GlobalKey@6a396b63 not found
  _ <- submitMustFail alice do exercise bobBazId BazFoo with c = alice, m = bob

  bobFooId2 <- submit bob do create Foo with m = bob, o = bob

  -- Fails because alice cannot see bobFooId.
  -- FIXME: Again a low-level error
  _ <- submitMustFail alice do exercise bobBazId BazFoo with c = alice, m = bob

  -- Disclose bobFooId to alice
  discloseId <- submit bob do create DiscloseFoo with s = bob, r = alice
  _ <- submit bob do exercise discloseId Disclose

  -- Now succeeds
  _ <- submit alice do exercise bobBazId BazFoo with c = alice, m = bob

  return ()

```

**Comment by bame-da** [comment]
I had a similar error:

```
"Scenario service backend error: BErrorClient (ClientIOError (GRPCIOBadStatusCode StatusUnknown (StatusDetails {unStatusDetails = \"\"})))"
```

In my case it occurs if a non-informee tries to do a lookupByKey on an existing key.

**Comment by ollie-da** [comment]
I get this also, using `digitalasset/daml-sdk:1.1.1`

**Comment by sofusmortensen** [comment]
(from slack)

`Scenario service backend error: BErrorClient (ClientIOError (GRPCIOBadStatusCode StatusUnknown (StatusDetails {unStatusDetails = \"\"})))`

That’s not a friendly way of telling me that the contract is not visible to bob when using `fetchByKey`.

`lookupByKey` is worse. If the contract doesn’t exist I get `None` - fine. If the contract does exist but I can’t see it - I get the error above (and indirectly information about its’ existence). I would assume to also get `None` for a contract that exists albeit isn’t visible to submitter.

**Comment by cocreature** [comment]
Fix is not in 1.2.0. Your example seems to work on the latest snapshot. See https://github.com/digital-asset/daml/commit/1c8691250d35da6e2759994ed8cd7d8e93094d2c. Github shows you two tags for each commit. The second one is the first that includes it afaik. I believe the first one is the latest but not quite sure.
